dev-tools/tests/test_omicron_dev.rs

```diff
@@ -289,66 +289,94 @@ async fn test_db_run() {
-        .expect("failed to join on connection")
-        .expect("connection failed with an error");
-    eprintln!("cleaned up connection");
+    let test_task = async {
+        let (client, connection) = dbrun
+            .listen_config
+            .connect(tokio_postgres::NoTls)
+            .await
+            .context("failed to connect to newly setup database")?;
+        let conn_task = tokio::spawn(connection);
+
+        anyhow::ensure!(
+            has_omicron_schema(&client).await,
+            "Missing Omicron schema"
+        );
+
+        // Now run db-populate.  It should fail because the database is already
+        // populated.
+        eprintln!("running db-populate");
+        let populate_result = Exec::cmd(&cmd_path)
+            .arg("db-populate")
+            .arg("--database-url")
+            .arg(&dbrun.listen_config_url)
+            .stdout(Redirection::Pipe)
+            .stderr(Redirection::Pipe)
+            .capture()
+            .context("failed to run db-populate")?;
+        eprintln!("exit status: {:?}", populate_result.exit_status);
+        eprintln!("stdout: {:?}", populate_result.stdout_str());
+        eprintln!("stdout: {:?}", populate_result.stderr_str());
+        anyhow::ensure!(
+            matches!(populate_result.exit_status, ExitStatus::Exited(1)),
+            "db-populate did not exit cleanly"
+        );
+        anyhow::ensure!(
+            populate_result
+                .stderr_str()
+                .contains("database \"omicron\" already exists"),
+            "db-populate did not fail when DB was already populated",
+        );
+        anyhow::ensure!(
+            has_omicron_schema(&client).await,
+            "Missing Omicron schema"
+        );
+
+        // Try again, but with the --wipe flag.
+        eprintln!("running db-populate --wipe");
+        let populate_result = Exec::cmd(&cmd_path)
+            .arg("db-populate")
+            .arg("--wipe")
+            .arg("--database-url")
+            .arg(&dbrun.listen_config_url)
+            .capture()
+            .context("failed to run db-populate")?;
+        anyhow::ensure!(
+            matches!(populate_result.exit_status, ExitStatus::Exited(0)),
+            "db-populate did not exit cleanly"
+        );
+        anyhow::ensure!(
+            has_omicron_schema(&client).await,
+            "Missing Omicron schema"
+        );
+
+        // Now run db-wipe.  This should work.
+        eprintln!("running db-wipe");
+        let wipe_result = Exec::cmd(&cmd_path)
+            .arg("db-wipe")
+            .arg("--database-url")
+            .arg(&dbrun.listen_config_url)
+            .capture()
+            .context("failed to run db-wipe")?;
+        anyhow::ensure!(
+            matches!(wipe_result.exit_status, ExitStatus::Exited(0)),
+            "db-wipe did not exit cleanly",
+        );
+        anyhow::ensure!(
+            !has_omicron_schema(&client).await,
+            "Missing Omicron schema"
```

Comment:
On all these `anyhow::ensure` calls, you don't have to provide a string.  If you don't, it will report the exact condition it was checking.  I personally prefer that.  If we want to keep the strings, then this one should be:

```suggestion
            "Unexpectedly found Omicron schema"
```

Reply:
Good idea, I'll leave the strings out.